### PR TITLE
Allow same native libraries if content is the same

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

Sometimes a user might end up with the same native library on the classpath, for example if the user depends on the uber-jar but another dependency directly depend on the classifier version. As long as the native libraries are the same there is no reason to fail loading it.

Modification:

- Check if the native library is the same (by compare the SHA256 of the file) and if so not fail
- Add a unit test

Result:

More robust check for native libraries

### CODE REVIEW

File: common/src/main/java/io/netty/util/internal/NativeLibraryLoader.java [hunk 1]
```diff
@@ -250,7 +252,51 @@ private static URL getResource(String path, ClassLoader loader) {
             case 1:
                 return urlsList.get(0);
             default:
-                throw new IllegalStateException("Multiple resources found for '" + path + "': " + urlsList);
+                try {
+                    MessageDigest md = MessageDigest.getInstance("SHA-256");
```
Comment (hyperxpro) [inline common/src/main/java/io/netty/util/internal/NativeLibraryLoader.java:256]: Why not MD5?

Comment (normanmaurer) [reply common/src/main/java/io/netty/util/internal/NativeLibraryLoader.java:256]: IMHO it doesn't really matter and md5 was flagged as insecure (which also not really matters tho)

Comment (hyperxpro) [reply common/src/main/java/io/netty/util/internal/NativeLibraryLoader.java:256]: Alright, that makes sense. :)

**Comment by normanmaurer** [comment]
PTAL @BuddhiWathsala

**Review by BuddhiWathsala — APPROVED** [review]
LGTM

**Pull request merged**